src/oscar/apps/analytics/receivers.py

```diff
@@ -4,16 +4,16 @@
 from django.db.models import F
 from django.dispatch import receiver
 
+from oscar.apps.basket.signals import basket_addition
+from oscar.apps.order.signals import order_placed
 from oscar.apps.search.signals import user_search
 from oscar.core.compat import user_is_authenticated
-from oscar.core.loading import get_class, get_classes
+from oscar.core.loading import get_classes
 
 UserSearch, UserRecord, ProductRecord, UserProductView = get_classes(
     'analytics.models', ['UserSearch', 'UserRecord', 'ProductRecord',
                          'UserProductView'])
 product_viewed = get_classes('catalogue.signals', ['product_viewed'])
```

Comment:
`product_viewed` also needs to be changed to direct import

Reply:
Overlooked this one, thanks!